### PR TITLE
Translate Usage - JsonAnalyzer

### PR DESCRIPTION
**Pull request by jwooley**

### CODE REVIEW

**Comment by codecrackerbuild** [comment]
Build passed! Details: https://ci.appveyor.com/project/code-cracker/code-cracker/build/1.0.0.420

File: src/VisualBasic/CodeCracker/Usage/JsonNetAnalyzer.vb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        Public Overrides ReadOnly Property SupportedDiagnostics As ImmutableArray(Of DiagnosticDescriptor)
+            Get
+                Return ImmutableArray.Create(Rule)
+            End Get
+        End Property
+
+        Public Overrides Sub Initialize(context As AnalysisContext)
+            context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "DeserializeObject", "Public Shared Overloads Function DeserializeObject(Of T)(value As String) As T"), SyntaxKind.InvocationExpression)
+            context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "Parse", "Public Shared Overloads Function Parse(json As String) As Newtonsoft.Json.Linq.JObject"), SyntaxKind.InvocationExpression)
+            'context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "Parse", "Nwetonsoft.Json.Linq.JArray.Parse(string)"), SyntaxKind.InvocationExpression)
```
Comment (giggio) [inline src/VisualBasic/CodeCracker/Usage/JsonNetAnalyzer.vb]: Why is this commented?

Comment (jwooley) [inline src/VisualBasic/CodeCracker/Usage/JsonNetAnalyzer.vb]: Sorry, it should have been removed. Both condition 2 and 3 are handled by 2 due to how VB translates the object differently from C# that is passied in the second parameter. I’ve removed it. Let me know if you need a new pull request.






Sent from Windows Mail





From: Giovanni Bassi
Sent: ‎Thursday‎, ‎February‎ ‎19‎, ‎2015 ‎10‎:‎16‎ ‎AM
To: 'code-cracker/code-cracker'
Cc: Jim Wooley





In src/VisualBasic/CodeCracker/Usage/JsonNetAnalyzer.vb:
> +            SupportedCategories.Usage,
> +            DiagnosticSeverity.Error,
> +            isEnabledByDefault:=True,
> +            description:=Description,
> +            helpLink:=HelpLink.ForDiagnostic(DiagnosticId.JsonNet))
> +
> +        Public Overrides ReadOnly Property SupportedDiagnostics As ImmutableArray(Of DiagnosticDescriptor)
> +            Get
> +                Return ImmutableArray.Create(Rule)
> +            End Get
> +        End Property
> +
> +        Public Overrides Sub Initialize(context As AnalysisContext)
> +            context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "DeserializeObject", "Public Shared Overloads Function DeserializeObject(Of T)(value As String) As T"), SyntaxKind.InvocationExpression)
> +            context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "Parse", "Public Shared Overloads Function Parse(json As String) As Newtonsoft.Json.Linq.JObject"), SyntaxKind.InvocationExpression)
> +            'context.RegisterSyntaxNodeAction(Sub(c) Analyze(c, "Parse", "Nwetonsoft.Json.Linq.JArray.Parse(string)"), SyntaxKind.InvocationExpression)


Why is this commented?

**Comment by codecrackerbuild** [comment]
Build passed! Details: https://ci.appveyor.com/project/code-cracker/code-cracker/build/1.0.0.421

**Pull request merged**